examples/slice/slice_utils.py

```diff
@@ -0,0 +1,341 @@
+import firedrake as fd
+import numpy as np
+from firedrake import op2
+
+
+def maximum(f):
```

Comment:
The `maximum` and `minimum` functions are probably general enough to have on their own in `utils` rather than specifically in the slice module